docs/vectors/c-source/sealbox_test_vectors.c

```diff
@@ -0,0 +1,150 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Test vector generator/checker for libsodium's box_seal APIs
+ * to build in a unix-like environment, use a command line like
+ * $ cc sealbox_test_vectors.c -I${IPATH} -L${LPATH} -lsodium -o sealbox_test_vectors
+ * with IPATH and LPATH defined to respectively point to libsodium's include path
+ * and to the directory containing the link library libsodium.a or libsodium.o
+ *
+ */
+#include <stdio.h>
+#include <string.h>
+#include <sodium.h>
+
+int checkone (char *hxsecret, char *hxpub, char *hxplaintext, char *hxencrypted) {
+
+    int pklen = crypto_box_PUBLICKEYBYTES;
+    int sklen = crypto_box_SECRETKEYBYTES;
+    int txtlen = strlen (hxplaintext) / 2;
+    int encrlen = strlen (hxencrypted) / 2;
+
+    char *skr = sodium_malloc (sklen);
+    char *pub = sodium_malloc (pklen);
+    char *txt = sodium_malloc (txtlen);
+    char *crpt = sodium_malloc (encrlen);
+    char *outp = sodium_malloc (txtlen);
+
+    int rs = sodium_hex2bin (skr, sklen, hxsecret, 2 * sklen,
+                             NULL, NULL, NULL);
+    rs |= sodium_hex2bin (pub, pklen, hxpub, 2 * pklen, NULL, NULL, NULL);
+    rs |= sodium_hex2bin (txt, txtlen, hxplaintext, strlen (hxplaintext),
+                          NULL, NULL, NULL);
+    rs |= sodium_hex2bin (crpt, encrlen, hxencrypted, strlen (hxencrypted),
+                          NULL, NULL, NULL);
+
+    if (rs == 0)
+        rs = crypto_box_seal_open (outp, crpt, encrlen, pub, skr);
+    if (rs == 0)
+        rs = sodium_memcmp (outp, txt, txtlen);
+
+    sodium_free (crpt);
+    sodium_free (txt);
+    sodium_free (skr);
+    sodium_free (pub);
+
+    return rs;
+}
+
+void gentestline (int minmsglen, int maxmsglen) {
+
+    int pklen = crypto_box_PUBLICKEYBYTES;
+    int sklen = crypto_box_SECRETKEYBYTES;
+    int txtlen = minmsglen + randombytes_uniform (maxmsglen - minmsglen + 1);
+    int encrlen = txtlen + crypto_box_SEALBYTES;
+
+    char *skr = sodium_malloc (sklen);
+    char *pub = sodium_malloc (pklen);
+    char *txt = sodium_malloc (txtlen);
+    char *crpt = sodium_malloc (encrlen);
+
+    crypto_box_keypair (pub, skr);
+    randombytes_buf (txt, txtlen);
+
+    crypto_box_seal (crpt, txt, txtlen, pub);
+
+    char *hskr = sodium_malloc (sklen * 2 + 1);
+    char *hpub = sodium_malloc (pklen * 2 + 1);
+    char *htxt = sodium_malloc (txtlen * 2 + 1);
+    char *hkrp = sodium_malloc (encrlen * 2 + 1);
+
+    sodium_bin2hex (hskr, sklen * 2 + 1, skr, sklen);
+    sodium_bin2hex (hpub, pklen * 2 + 1, pub, pklen);
+    sodium_bin2hex (htxt, txtlen * 2 + 1, txt, txtlen);
+    sodium_bin2hex (hkrp, encrlen * 2 + 1, crpt, encrlen);
+
+    printf ("%s\t%s\t%s\t%s\n", hskr, hpub, htxt, hkrp);
+}
+
+int main (int argc, char **argv) {
+/*
+ * If called without any argument, the resulting executable will
```

Comment:
I tried compiling libsodium and sealbox_test_vectors.c on fresh install of Ubuntu 16.04 x64.

    sudo apt update && sudo apt upgrade --yes
    wget https://download.libsodium.org/libsodium/releases/libsodium-1.0.12.tar.gz
    tar -xzf libsodium-1.0.12.tar.gz
    cd libsodium-1.0.12/
    ./configure
    make && make check
    sudo make install
    wget https://raw.githubusercontent.com/lmctv/pynacl/document-sealedbox-vectors/docs/vectors/c-source/sealbox_test_vectors.c
    cc -o sealbox_test_vectors -l sodium sealbox_test_vectors.c

The last command produces following errors:

    /tmp/cccv8Irx.o: In function `checkone':
    sealbox_test_vectors.c:(.text+0x53): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x64): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x75): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x86): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x97): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0xcb): undefined reference to `sodium_hex2bin'
    sealbox_test_vectors.c:(.text+0x102): undefined reference to `sodium_hex2bin'
    sealbox_test_vectors.c:(.text+0x140): undefined reference to `sodium_hex2bin'
    sealbox_test_vectors.c:(.text+0x17e): undefined reference to `sodium_hex2bin'
    sealbox_test_vectors.c:(.text+0x1ac): undefined reference to `crypto_box_seal_open'
    sealbox_test_vectors.c:(.text+0x1ce): undefined reference to `sodium_memcmp'
    sealbox_test_vectors.c:(.text+0x1dd): undefined reference to `sodium_free'
    sealbox_test_vectors.c:(.text+0x1e9): undefined reference to `sodium_free'
    sealbox_test_vectors.c:(.text+0x1f5): undefined reference to `sodium_free'
    sealbox_test_vectors.c:(.text+0x201): undefined reference to `sodium_free'
    /tmp/cccv8Irx.o: In function `gentestline':
    sealbox_test_vectors.c:(.text+0x232): undefined reference to `randombytes_uniform'
    sealbox_test_vectors.c:(.text+0x252): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x263): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x274): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x285): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x29c): undefined reference to `crypto_box_keypair'
    sealbox_test_vectors.c:(.text+0x2b1): undefined reference to `randombytes_buf'
    sealbox_test_vectors.c:(.text+0x2cb): undefined reference to `crypto_box_seal'
    sealbox_test_vectors.c:(.text+0x2dd): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x2f3): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x309): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x31f): undefined reference to `sodium_malloc'
    sealbox_test_vectors.c:(.text+0x344): undefined reference to `sodium_bin2hex'
    sealbox_test_vectors.c:(.text+0x365): undefined reference to `sodium_bin2hex'
    sealbox_test_vectors.c:(.text+0x386): undefined reference to `sodium_bin2hex'
    sealbox_test_vectors.c:(.text+0x3a7): undefined reference to `sodium_bin2hex'
    /tmp/cccv8Irx.o: In function `main':
    sealbox_test_vectors.c:(.text+0x3fb): undefined reference to `sodium_init'
    collect2: error: ld returned 1 exit status

So I'm probably not linking libsodium properly. Should the cc command specify sodium.h or am I skipping a step?